GitCommands/Git/GitItemStatus.cs

```diff
@@ -21,9 +30,30 @@ public class GitItemStatus : IComparable<GitItemStatus>
         public bool IsConflict { get; set; }
         public bool IsAssumeUnchanged { get; set; }
         public bool IsSkipWorktree { get; set; }
-        public bool IsStaged { get; set; } = true;
         public bool IsSubmodule { get; set; }
         public string RenameCopyPercentage { get; set; }
+        private StagedStatus StagedUnsafe { get; set; } = StagedStatus.Unknown;
+        public StagedStatus Staged
+        {
+            get
+            {
+                // Staged is not available in all scenarios, this is temporary (?) to catch usage
+                if (StagedUnsafe == StagedStatus.Unknown)
+                {
+#if DEBUG
+                    Debug.Assert(false, "StagedStatus is not set - this throws in Release builds. Continue should generally be OK.");
+#else
+                    throw new ArgumentException("StagedStatus is not set", StagedUnsafe.ToString());
```

Comment:
what would it mean in release?

Reply:
I probably want to remove the throw in a final release. 
This throw will occur when Staged is used without being set. It can potentially occur in a few situations, even if most usage should be OK. _if_ there is some key scenario left, an alpha may raise this issue so it can be handled or ignored.

The previous code just set IsStaged to true/false and the usage of the field was a little random - OK in some situations but not other.

Reply:
Let's not throw, the last thing we want is to crash the app.
If for some reason _Staged_ is unset and can't remain as such, should we treat it as _unstaged_?